core/client/src/main/java/alluxio/client/block/AlluxioBlockStore.java

```diff
@@ -40,20 +44,37 @@
 public final class AlluxioBlockStore {
   private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
 
-  private static AlluxioBlockStore sClient = null;
+  // Only one block store will be kept for each master.
+  private static final Map<InetSocketAddress, AlluxioBlockStore> CACHED_BLOCK_STORES =
```

Comment:
Sorry, by breaking it up I thought you meant decentralizing it so that the objects which use `AlluxioBlockStore` each have their own `AlluxioBlockStore` vs holding all of the `AlluxioBlockStores` in a static map in this class.